addons/sale_async_emails/models/sale_order.py

```diff
@@ -0,0 +1,45 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.tools.misc import str2bool
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    pending_email_template_id = fields.Many2one(
+        string="Pending Email Template",
+        help="The template of the pending email that must be sent asynchronously.",
+        comodel_name='mail.template',
+        ondelete='set null',
+        readonly=True,
+    )
+
+    def _send_order_notification_mail(self, mail_template):
+        """ Override of `sale` to reschedule order status emails to be sent asynchronously. """
+        async_send = str2bool(self.env['ir.config_parameter'].sudo().get_param('sale.async_emails'))
+        cron = self.env.ref('sale_async_emails.cron', raise_if_not_found=False)
```

Comment:
Could be nice to avoid the existence check for the cron here when the config param is false.